xonsh/parsers/base.py

```diff
@@ -1894,15 +1895,16 @@ def p_testlist_test(self, p):
         """testlist : test"""
         p1 = p[1]
         if isinstance(p1, ast.Tuple) and (hasattr(p1, '_real_tuple') and
-                                          p1._real_tuple):
+                                          p1._real_tuple and p1.elts):
             p1.lineno, p1.col_offset = lopen_loc(p1.elts[0])
         p[0] = p1
 
     def p_testlist_single(self, p):
         """testlist : test COMMA"""
         p1 = p[1]
-        if isinstance(p1, ast.Tuple) and (hasattr(p1, '_real_tuple') and
-                                          p1._real_tuple):
+        if isinstance(p1, ast.List) or (isinstance(p1, ast.Tuple) and
+                                        hasattr(p1, '_real_tuple') and
+                                        p1._real_tuple):
```

Comment:
I'd prefer `getattr(p1, '_real_tuple', False)`, but still, I think that style changes might be outside the scope of this particular PR.  Though do feel free to fix things up as you go, @blahgeek, if your want!